fhir-search/src/main/java/com/ibm/fhir/search/parameters/ParametersUtil.java

```diff
@@ -79,185 +64,95 @@
     private static final String COMMA = ",";
     private static final String EQUALS = "=";
 
-    // Unsupported Operations in FHIR Path
-    public static final String OPERATION_RESOLVE = "resolve()";
-    public static final String OPERATION_CONTAINS = ".contains";
-    private static final List<String> UNSUPPORTED_OPERATIONS = Arrays.asList(OPERATION_CONTAINS);
-
-    private static final String REPLACE_RESOLVE = "(\\.where\\([\\s]{0,}resolve\\(\\)[\\s]{0,}[is]{0,2}[\\s]{0,}[\\w]{0,}\\))";
+    private static final Map<String, ParametersMap> builtInSearchParameters = loadBuiltIns();
 
-    private static final String CONTAINS_EXPR = "\\.contains\\.";
-    private static final String CONTAINS_EXPR_REPLACE = ".`contains`.";
-    private static final String CONTAINS_END_EXPR = ".contains";
-    private static final String CONTAINS_END_EXPR_REPLACE = ".`contains`";
-
-    private static final Map<String, Map<String, SearchParameter>> builtInSearchParameters = loadBuiltIn();
-
-    private static final String INVALID_EXPRESSION = "/NONE/";
+    private static final String MISSING_EXPRESSION = "/NONE/";
 
     private ParametersUtil() {
         // No Operation
     }
 
     /**
-     * Loads the class in the classloader to initialize static members. Call this before using the class in order to
-     * avoid a slight performance hit on first use.
+     * Loads the class in the classloader to initialize static members.
+     * Call this before using the class in order to avoid a slight performance hit on first use.
      */
     public static void init() {
         // No Operation
     }
 
-    /*
-     * wraps the error thrown, catches it, and enables a user to operate.
-     * @return
-     */
-    private static Map<String, Map<String, SearchParameter>> loadBuiltIn() {
-        Map<String, Map<String, SearchParameter>> result = null;
-        try {
-            result = populateSearchParameterMapFromResource(FHIR_DEFAULT_SEARCH_PARAMETERS_FILE);
-        } catch (IOException e) {
-            // This Exception is HIGHLY improbable.
-            log.warning(BUILTIN_ERROR_EXCEPTION);
-        }
-        return result;
-    }
-
     /**
-     * This is a convenience function that simply returns the built-in (spec-defined) SearchParameters as a map keyed by
-     * resource type.
+     * Loads the built-in search parameters and constructs . 
+     * 
+     * @return a map of ParametersMaps, keyed by resourceType
+     * @throws FHIRParserException 
```

Comment:
hum... do you actually throw this?  note the signature is not showing it.

Reply:
good catch, will remove

Reply:
done